### PR TITLE
PBI-03 Submitting Assignments

### PR DESCRIPTION
**Pull request by johaneschristian**
Create end-point to submit and save assignment attempt. This commit integrates One-Day-Intern application with the google-storage (bucket) backing services that serves as the storage server. The credentials will be provided as an environment variable in the Cloud Run. At this current PR, the mechanism for validating whether an assessee is still eligible for submitting an assignment **has not been implemented** and will be implemented after the downloading end-point is completed.

### CODE REVIEW

File: assessment/models.py [hunk 1]
```diff
@@ -426,3 +421,89 @@ class VideoConferenceRoomSerializer(serializers.ModelSerializer):
...
+class ToolAttempt(PolymorphicModel):
+    tool_attempt_id = models.UUIDField(default=uuid.uuid4, primary_key=True)
+    grade = models.FloatField(default=0)
+    test_flow_attempt = models.ForeignKey('assessment.TestFlowAttempt', on_delete=models.CASCADE)
+    assessment_tool_attempted = models.ForeignKey('assessment.AssessmentTool', on_delete=models.CASCADE, default=None)
+
+
+class AssignmentAttempt(ToolAttempt):
+    file_upload_directory = models.TextField(null=True)
+    filename = models.TextField(default=None, null=True)
```
Comment (rashadaziz) [inline assessment/models.py:435]: Hello jo, just wondering, will this property be responsible for persisting the originally submitted filename? If so, how do we handle cases where two assesses submitted files with identical names? Let me know :D

File: assessment/services/google_storage.py [hunk 2]
```diff
@@ -11,4 +12,7 @@ def setup_google_storage_credentials():
 
 
 def upload_file_to_google_bucket(destination_file_name, bucket_name, file):
-    pass
+    storage_client = storage.Client()
+    bucket = storage_client.get_bucket(bucket_name)
+    blob = bucket.blob(destination_file_name)
+    blob.upload_from_file(file_obj=file, rewind=True)
```
Comment (whenwe177) [inline assessment/services/google_storage.py:18]: Hello, Jo. If I may ask, the functionality of uploading blobs are built in from the Google Cloud package, yes?

Comment (johaneschristian) [reply assessment/services/google_storage.py:18]: Hello, Alex. Yes, the blobs are built in from the Google cloud module.

Comment (whenwe177) [reply assessment/services/google_storage.py:18]: Okay thanks Jo. This means I need to install the package when I pull from staging.

File: assessment/models.py (hunk 1 above)
Comment (johaneschristian) [reply assessment/models.py:435]: Hello, Rashad. Our current implementation overwrites the previous submission when a new file is submitted. In doing so, as shown in the assessment_event_attempt service module, both the cloud directory and filename will be updated in accordance to the new submitted file.

**Pull request merged**